packages/core/src/RenderingEngine/helpers/setDefaultVolumeVOI.ts

```diff
@@ -0,0 +1,234 @@
+   * for the scaled PT to be 0, 5
+   */
+  if (generalSeriesModule.modality === 'PT' && imageVolume.isPrescaled) {
+    return {
+      lower: 0,
+      upper: 5,
+    };
+  }
+
+  return voi;
+}
+
+/**
+ * Get the VOI from the metadata of the middle slice of the image volume. It checks
+ * the metadata for the VOI and if it is not found, it returns null
+ *
+ * @param imageVolume - The image volume that we want to get the VOI from.
+ * @returns VOIRange with lower and upper values
+ */
+function getVOIFromMetadata(imageVolume: IImageVolume): VOIRange {
+  const { imageIds } = imageVolume;
+
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageIds[imageIdIndex];
+
+  const voiLutModule = metaData.get('voiLutModule', imageId);
+
+  if (voiLutModule && voiLutModule.windowWidth && voiLutModule.windowCenter) {
+    const { windowWidth, windowCenter } = voiLutModule;
+
+    const voi = {
+      windowWidth: Array.isArray(windowWidth) ? windowWidth[0] : windowWidth,
+      windowCenter: Array.isArray(windowCenter)
+        ? windowCenter[0]
+        : windowCenter,
+    };
+
+    const { lower, upper } = windowLevel.toLowHighRange(
+      Number(voi.windowWidth),
+      Number(voi.windowCenter)
+    );
+
+    return {
+      lower,
+      upper,
+    };
+  }
+}
+
+/**
+ * It loads the middle slice image (middle imageId) and based on its min
+ * and max pixel values, it calculates the VOI.
+ *
+ * @param imageVolume - The image volume that we want to get the VOI from.
+ * @returns The VOIRange with lower and upper values
+ */
+async function getVOIFromMinMax(imageVolume: IImageVolume): Promise<VOIRange> {
+  const { scalarData, imageIds } = imageVolume;
+
+  // Get the middle image from the list of imageIds
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageVolume.imageIds[imageIdIndex];
+  const generalSeriesModule =
+    metaData.get('generalSeriesModule', imageId) || {};
+  const { modality } = generalSeriesModule;
+  const modalityLutModule = metaData.get('modalityLutModule', imageId) || {};
+
+  const numImages = imageIds.length;
+  const bytesPerImage = scalarData.byteLength / numImages;
+  const length = scalarData.length / numImages;
+
+  let type;
+
+  if (scalarData instanceof Uint8Array) {
+    type = 'Uint8Array';
+  } else if (scalarData instanceof Float32Array) {
+    type = 'Float32Array';
+  } else {
+    throw new Error('Unsupported array type');
+  }
+
+  const scalingParameters: ScalingParameters = {
+    rescaleSlope: modalityLutModule.rescaleSlope,
+    rescaleIntercept: modalityLutModule.rescaleIntercept,
+    modality,
+  };
+
+  let scalingParametersToUse;
+  if (modality === 'PT') {
+    const suvFactor = metaData.get('scalingModule', imageId);
+
+    if (suvFactor) {
+      scalingParametersToUse = {
+        ...scalingParameters,
+        suvbw: suvFactor.suvbw,
+      };
+    }
+  }
+
+  const options = {
+    targetBuffer: {
+      arrayBuffer: scalarData.buffer,
```

Comment:
In `_getImageScalarDataFromImageVolume` you do a bunch of calculations again, but I think you more or less already have all the variables you need here to get the array view you need right?